src/send_kcidb.py

```diff
@@ -8,126 +8,116 @@
 # Copyright (C) 2022 Collabora Limited
 # Author: Jeny Sadadia <jeny.sadadia@collabora.com>
 
+"""KCIDB bridge service"""
+
 import datetime
-import logging
 import os
 import sys
-import traceback
 
 import kernelci
 import kernelci.db
-from kernelci.config import load
+import kernelci.config
 from kernelci.cli import Args, Command, parse_opts
 from kcidb import Client
 import kcidb
 
-from logger import Logger
+from base import Service
 
 
-class cmd_run(Command):
-    help = "Listen for events and send them to KCDIB"
-    args = [Args.db_config]
+class KCIDBBridge(Service):
+    def __init__(self, configs, args, name):
+        super().__init__(configs, args, name)
+        self._kcidb_topic_name = args.kcidb_topic_name
+        self._kcidb_project_id = args.kcidb_project_id
+        self._client = None
 
-    def __init__(self, sub_parser, name):
-        super().__init__(sub_parser, name)
-        self._logger = Logger("config/logger.conf", "send_kcidb")
+    def _kcidb_connect(self):
+        return Client(project_id=self._kcidb_project_id,
+                      topic_name=self._kcidb_topic_name)
```

Comment:
I mean, the `Client` object could be created directly in `_setup()`.  Also the KCIDB topic and project ID could be read from the `args` there rather than storing them as instance variables in the object.  That's a bit more of a grey area but I think it's better to set everything up in `_setup()` as much as possible.

Reply:
Having separate functions increases the code's reusability.
That's why I created the `kcidb_connect` method and also verified if the connection is successful in the method.
Since the method just has a single line of code now, it won't make much of a difference and the method can be dropped.